build-packages/webpack-i18n-runtime/build-config/webpack-i18n-tracker/localization-manager/index.js

```diff
@@ -0,0 +1,208 @@
+                // Handle missing translation
+                if (!this.translationMap[localeCode].merged[usedTranslation]) {
+                    // Handle first missing translation for locale
+                    if (!missingTranslations[localeCode]) {
+                        missingTranslations[localeCode] = new Set();
+                    }
+
+                    missingTranslations[localeCode].add(usedTranslation);
+                }
+            }
+        }
+
+        // Ignore the default locale
+        delete missingTranslations[this.defaultLocale];
+
+        // Count the missing translations for each locale
+        const missingTranslationsCountMap = Object.fromEntries(Object.entries(missingTranslations).map(
+            ([localeCode, missingTranslations]) => [localeCode, missingTranslations.size]
+        ));
+
+        // Handle OK case
+        if (!Object.keys(missingTranslationsCountMap).length) {
+            return;
+        }
+
+        // 1. log which ones are missing
+        afterEmitLogger.logMessage(missingTranslationsMessage(missingTranslationsCountMap));
+
+        // 2. Update files contents
+        appendMissingTranslationsToFiles(missingTranslations);
+    }
+
+    /**
+     * Handle unused translations
+     */
+    handleUnusedTranslations() {
+        const unusedTranslations = {};
+
+        // Loop over translations from files
+        for (const localeCode in this.translationMap) {
+            // Check each translation from there
+            for (const translatable in this.translationMap[localeCode].child) {
+                // Handle translation not used in the application
+                if (!this.usedTranslations.has(translatable)) {
+                    // Handle first unused for the locale
+                    if (!unusedTranslations[localeCode]) {
+                        unusedTranslations[localeCode] = new Set();
+                    }
+
+                    unusedTranslations[localeCode].add(translatable);
+                }
+            }
+        }
+
+        // Handle OK case
+        if (!Object.keys(unusedTranslations).length) {
+            return;
+        }
+
+        // log which ones are unused
+        afterEmitLogger.logMessage(unusedTranslationsMessage(unusedTranslations));
+    }
+
+    handleOverriddenEmptyTranslations() {
+        if (!Object.keys(this.overriddenEmptyTranslations).length) {
+            return;
+        }
+
+        afterEmitLogger.logMessage(emptyTranslationsOverriddenMessage(this.overriddenEmptyTranslations));
+    }
+
+    handleOverridingEmptyTranslation(localeCode, translatable, incomingValue) {
+        if (!this.overriddenEmptyTranslations[localeCode]) {
+            this.overriddenEmptyTranslations[localeCode] = {};
+        }
+
+        this.overriddenEmptyTranslations[localeCode][translatable] = incomingValue;
+    }
+
+    handleNonExtractableParam(resource) {
+        this.nonExtractableCases.push(resource);
```

Comment:
Is this a handler for non string translation? Like if user tries to pass in the variable? Because previously it generated `null`. Can we handle this?

Reply:
It indeed is. Now, `null` is not generated.